ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/RemoteConnectorExecutor.java

```diff
@@ -188,16 +200,78 @@ && getUserRateLimiterMap().get(user.getName()) != null
             if (getMlGuard() != null && !getMlGuard().validate(payload, MLGuard.Type.INPUT)) {
                 throw new IllegalArgumentException("guardrails triggered for user input");
             }
-            invokeRemoteModel(mlInput, parameters, payload, tensorOutputs, countDownLatch, actionListener);
+            if (getConnectorClientConfig().getMaxRetryTimes() != 0) {
```

Comment:
Why checking max retry times instead of a switch? We should consider to add a switch to control retry since some users might not want to enable this feature.

Reply:
Set the retry times to zero is same as switch off the feature. We can point out this in our documents. 

We already have 4 settings for retry and don't want it over designed